File: lib/core/env.dart
/// Compile-time environment configuration.
///
/// Pass values via `--dart-define` when running/building, e.g.:
/// flutter run --dart-define=SUPABASE_URL=... --dart-define=SUPABASE_ANON_KEY=...
class Env {
  const Env._();

  static const String supabaseUrl = String.fromEnvironment('SUPABASE_URL');
  static const String supabaseAnonKey =
      String.fromEnvironment('SUPABASE_ANON_KEY');

  static bool get isConfigured =>
      supabaseUrl.isNotEmpty && supabaseAnonKey.isNotEmpty;
}

File: src/core/env.ts
/**
 * Compile-time configuration. Values come from `EXPO_PUBLIC_*` env vars,
 * which Expo automatically inlines at bundle time.
 *
 * Set them in `.env` (gitignored) for local dev or via `eas secret` for
 * production builds.
 */
export const Env = {
  supabaseUrl: process.env.EXPO_PUBLIC_SUPABASE_URL ?? '',
  supabaseAnonKey: process.env.EXPO_PUBLIC_SUPABASE_ANON_KEY ?? '',
} as const;

export const supabaseConfigured =
  Env.supabaseUrl.length > 0 && Env.supabaseAnonKey.length > 0;
